lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -182,6 +182,15 @@ pub fn parse_token(stream: ParseStream) -> ParseResult<ConstraintToken> {
             };
             ConstraintToken::Bump(Context::new(ident.span(), ConstraintTokenBump { bump }))
         }
+        "program_id" => {
```

Comment:
I think the other name that the solana sdk uses for this is `base`. I don't have a preference though.